edb/pgsql/ast.py

```diff
@@ -1005,3 +1005,87 @@ class IteratorCTE(ImmutableBase):
     path_id: irast.PathId
     cte: CommonTableExpr
     parent: typing.Optional[IteratorCTE]
+
+
+class Statement(Base):
```

Comment:
https://github.com/edgedb/edgedb/blob/9a396dcd76d1b8bbf8c78c2b42d7891b0aaa27bf/edb/pgsql/parser/__init__.py#L30

This would return `List[pgast.Base]` instead? Or perhaps `List[Union[pgast.Query, pgast.Statement]]`?

Reply:
Yep, you are right - type annotations inside ast_builder are more like suggestions than hard requirements, I guess.